interface/batchcom/emailnotification.php

```diff
@@ -28,52 +28,44 @@
-$next_app_date = date("Y-m-d");
-$hour = "12";
-$min = "15";
-$provider_name = "EMR Group";
-$message = "Welcome to EMR Group";
-$type = "Email";
-$email_sender = "EMR Group";
-$email_subject = "Welcome to EMR Group";
 // process form
 if ($_POST['form_action'] == 'save') {
     if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
         CsrfUtils::csrfNotVerified();
     }
 
-    //validation uses the functions in notification.inc.php
-    if ($_POST['email_sender'] == "") {
-        $form_err .= xl('Empty value in "Email Sender"') . '<br />';
+    if (! is_numeric($_POST['notification_id'])) {  // shouldn't happen
+        $form_err .= xl('Missing/invalid notification id') . '<br />';
     }
 
-    if ($_POST['email_subject'] == "") {
-        $form_err .= xl('Empty value in "Email Subject"') . '<br />';
+    if (empty($_POST['email_sender'])) {
+        $form_err .= xl('Empty value in "Email Sender"') . '<br />';
     }
 
-    //validate dates
-    if (!check_date_format($_POST['next_app_date'])) {
-        $form_err .= xl('Date format for "Next Appointment" is not valid') . '<br />';
+    if (empty($_POST['email_subject'])) {
+        $form_err .= xl('Empty value in "Email Subject"') . '<br />';
     }
 
-    // validates and or
-    if ($_POST['provider_name'] == "") {
+    if (empty($_POST['provider_name'])) {
         $form_err .= xl('Empty value in "Name of Provider"') . '<br />';
     }
 
-    if ($_POST['message'] == "") {
+    if (empty($_POST['message'])) {
         $form_err .= xl('Empty value in "Email Text"') . '<br />';
     }
 
-    //process sql
+    // Store the new settings.  sms_gateway_type is not used for email.
+    // next_app_time, next_app_date, and notification_sent_date don't appear
+    // to be used by anyone.  notification_id is the pk, and should always
+    // be 2 for email settings.
+
     if (!$form_err) {
-        $next_app_time = $_POST['hour'] . ":" . $_POST['min'];
-        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type` ) ";
-        $sql_value = " (?, ?, ?, ?, ?, ?, ?, ?, ?) ";
-        $values = array($_POST['notification_id'], $_POST['sms_gateway_type'], $_POST['next_app_date'], $next_app_time,
-                        $_POST['provider_name'], $_POST['message'], $_POST['email_sender'], $_POST['email_subject'],
-                        $type);
+        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type`, `notification_sent_date` ) ";
+        $sql_value = " (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ";
+        $values = array($_POST['notification_id'], '', '0000-00-00', ':',
+                        $_POST['provider_name'], $_POST['message'],
+                        $_POST['email_sender'], $_POST['email_subject'],
+                        'Email', '2007-09-30 00:00:00');
```

Comment:
why the 2007 date here?

Reply:
The old code didn't put any value there at all, which caused the update to be rejected since that column is NOT NULL.  I just used the value that's seeded by sql/database.sql so it would work, but honestly that column doesn't appear to be used anywhere, so I'd be happier nuking it entirely.  I'm pretty new to openemr, though, and I'm trying not to break things (at least not RIGHT away).